<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Klara Kaleb" />

<meta name="date" content="2017-10-12" />

<title>Get started with Roma</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Get started with Roma</h1>
<h4 class="author"><em>Klara Kaleb</em></h4>
<h4 class="date"><em>2017-10-12</em></h4>



<p>This little vignette shows you how to get started with the <code>roma</code> package. roma is a wrapper for the REST API for the Orthologous MAtrix project (OMA) which is a database for the inference of orthologs among complete genomes.</p>
<p>For more details on the OMA project, see <a href="https://omabrowser.org/oma/home/" class="uri">https://omabrowser.org/oma/home/</a> .</p>
<div id="some-useful-functions" class="section level2">
<h2>Some useful functions</h2>
<p>The package contains a range of functions that are used to query the database in an R friendly way. This vignette describes some of them, whereas the rest are described in more detail in other vignettes.</p>
<div id="getxref" class="section level3">
<h3>getXref</h3>
<p>This function searches the OMA database for entries containing the pattern defined and returns the results in a dataframe. Hence, it is usually a good starting place. Sample code is below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(roma)

results &lt;-<span class="st"> </span><span class="kw">getXref</span>(<span class="dt">pattern=</span><span class="st">&quot;MAL&quot;</span>)

results</code></pre></div>
<pre><code>## NULL</code></pre>
</div>
<div id="getgenomealignment" class="section level3">
<h3>getGenomeAlignment</h3>
<p>This function serves to obtain the orthologs for 2 whole genomes. The result is a dataframe containing information on each member in the pair and their relationship.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(roma)

alignment &lt;-<span class="st"> </span><span class="kw">getGenomeAlignment</span>(<span class="dt">genome_id1 =</span> <span class="st">&quot;YEAST&quot;</span>, <span class="dt">genome_id2 =</span> <span class="st">&quot;ASHGO&quot;</span>)

<span class="kw">head</span>(alignment)</code></pre></div>
<pre><code>##   entry_1.entry_nr                                entry_1.entry_url
## 1          6567568 https://omadev.cs.ucl.ac.uk/api/protein/6567568/
## 2          6567157 https://omadev.cs.ucl.ac.uk/api/protein/6567157/
## 3          6567158 https://omadev.cs.ucl.ac.uk/api/protein/6567158/
## 4          6567565 https://omadev.cs.ucl.ac.uk/api/protein/6567565/
## 5          6566009 https://omadev.cs.ucl.ac.uk/api/protein/6566009/
## 6          6566007 https://omadev.cs.ucl.ac.uk/api/protein/6566007/
##   entry_1.omaid entry_1.canonicalid             entry_1.sequence_md5
## 1    YEAST04806          RCE1_YEAST 605098a0697ad8fc7af2101e758033cb
## 2    YEAST04395          ZDS2_YEAST 8cc75f16fbfd321833abc48fd1173154
## 3    YEAST04396          YMK8_YEAST 783fea3b573632292d89a5a5218b6e90
## 4    YEAST04803          SCS7_YEAST 9307c7a6e80ed39d8b6529329fee1819
## 5    YEAST03247          SMC3_YEAST 4e8f1295434b44ae8f749cad976b966e
## 6    YEAST03245          NET1_YEAST f2ba71aea520ea66f015ba357eb6e8c6
##   entry_1.oma_group entry_1.oma_hog_id entry_1.chromosome
## 1            590938     HOG:0372772.4c               XIII
## 2            219496     HOG:0189085.1b               XIII
## 3            212453        HOG:0188799               XIII
## 4            750461  HOG:0376737.2c.3b               XIII
## 5            484612     HOG:0385156.3a                  X
## 6            512189     HOG:0188664.2b                  X
##   entry_1.locus.start entry_1.locus.end entry_1.locus.strand
## 1              814364            815311                   -1
## 2               51640             54468                    1
## 3               54793             55110                    1
## 4              809623            810777                   -1
## 5              299157            302849                   -1
## 6              295245            298814                    1
##   entry_2.entry_nr                                entry_2.entry_url
## 1          6548024 https://omadev.cs.ucl.ac.uk/api/protein/6548024/
## 2          6548025 https://omadev.cs.ucl.ac.uk/api/protein/6548025/
## 3          6548026 https://omadev.cs.ucl.ac.uk/api/protein/6548026/
## 4          6548027 https://omadev.cs.ucl.ac.uk/api/protein/6548027/
## 5          6548028 https://omadev.cs.ucl.ac.uk/api/protein/6548028/
## 6          6548029 https://omadev.cs.ucl.ac.uk/api/protein/6548029/
##   entry_2.omaid entry_2.canonicalid             entry_2.sequence_md5
## 1    ASHGO00001              Q75FB7 a9b1a6dc9afb2b02afe8fdf8029b5f22
## 2    ASHGO00002              Q75FB6 5d186037c4dd0a89b34d70d596fac86d
## 3    ASHGO00003              Q75FB5 3a83b276f0c9034f7cf66277e7d6c983
## 4    ASHGO00004              Q75FB4 a8611c3f24ac6599e6a36a2710f6d24a
## 5    ASHGO00005              Q75FB3 07277f0ca66fcb49d175667272f1547f
## 6    ASHGO00006              Q75FB2 2b666569856af5f068e532ade89c1140
##   entry_2.oma_group entry_2.oma_hog_id entry_2.chromosome
## 1            590938     HOG:0372772.4c                  I
## 2            219496        HOG:0189085                  I
## 3            212453        HOG:0188799                  I
## 4            750461  HOG:0376737.2c.3b                  I
## 5            484612     HOG:0385156.3a                  I
## 6            512189     HOG:0188664.3b                  I
##   entry_2.locus.start entry_2.locus.end entry_2.locus.strand rel_type
## 1                8108              9067                    1      1:1
## 2                9537             12593                    1      1:1
## 3               12906             13244                    1      1:1
## 4               13713             14846                    1      1:1
## 5               16155             19850                    1      1:1
## 6               20056             23721                   -1      1:1
##   distance   score
## 1 122.0000  636.04
## 2  95.0000 1424.24
## 3  50.0000  557.51
## 4  37.7442 2682.46
## 5  58.0000 5548.40
## 6  90.0000 1844.35</code></pre>
</div>
<div id="getdata" class="section level3">
<h3>getData</h3>
<p>This master-function function serves to obtain the information for a single entry in a database - either a group, protein or a genome. The data type is specified by setting the “type” argument and a specific entry by its ID - below are the possible ID’s for different object type.</p>
<ul>
<li>Protein
<ul>
<li>oma id</li>
<li>canonical id</li>
<li>entry number - release specific</li>
</ul></li>
<li>Genome
<ul>
<li>NCBI taxon id</li>
<li>UniProt species code</li>
</ul></li>
<li>Group
<ul>
<li>group number</li>
<li>group fingerprint</li>
<li>id of one of its members</li>
</ul></li>
</ul>
<p>The result is an S3 Object, attributes of which can be accessed via $. Below is an example of obtaining information for a specific OMA group.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(roma)

group &lt;-<span class="st"> </span><span class="kw">getData</span>(<span class="dt">type=</span><span class="st">&quot;Group&quot;</span>,<span class="dt">id=</span><span class="st">&quot;YEAST58&quot;</span>)

group<span class="op">$</span>fingerprint</code></pre></div>
<pre><code>## [1] &quot;PNDKFPS&quot;</code></pre>
</div>
<div id="resolveurl" class="section level3">
<h3>resolveURL</h3>
<p>In most cases there is great quantity of information available for a given entry and this impacts the data retrival time. Due to this, the information available for such entries in split into a number of endpoints and these are included appropriatelly as redirects. This function allows the user to obtain further information behind those urls.</p>
</div>
<div id="formatdata" class="section level3">
<h3>formatData</h3>
<p>When an object is created from the corresponding response to a request, its data is returned in the simplest most form to speed up loading time. This sometimes results in very long named lists which can be a challenge to read - the purpose of this function is to format those lists in an appropriate dataframe.</p>
<p>One can see some of the above functions in action in the below example of trying to obtain the list of orthologs for a protein entry.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(roma)

protein &lt;-<span class="st"> </span><span class="kw">getData</span>(<span class="dt">type=</span><span class="st">&quot;Protein&quot;</span>,<span class="dt">id=</span><span class="st">&quot;YEAST58&quot;</span>)

<span class="kw">attributes</span>(protein)</code></pre></div>
<pre><code>## $names
##  [1] &quot;entry_nr&quot;        &quot;entry_url&quot;       &quot;omaid&quot;          
##  [4] &quot;canonicalid&quot;     &quot;sequence_md5&quot;    &quot;oma_group&quot;      
##  [7] &quot;oma_hog_id&quot;      &quot;chromosome&quot;      &quot;locus&quot;          
## [10] &quot;roothog_id&quot;      &quot;hog_levels&quot;      &quot;sequence_length&quot;
## [13] &quot;sequence&quot;        &quot;cdna&quot;            &quot;domains&quot;        
## [16] &quot;xref&quot;            &quot;orthologs&quot;       &quot;ontology&quot;       
## [19] &quot;oma_group_url&quot;   &quot;oma_hog_members&quot;
## 
## $class
## [1] &quot;OMAObject&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">protein<span class="op">$</span>orthologs</code></pre></div>
<pre><code>## [1] &quot;https://omadev.cs.ucl.ac.uk/api/protein/6562820/orthologs/&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">orthologs =<span class="st"> </span><span class="kw">resolveURL</span>(protein<span class="op">$</span>orthologs)

<span class="kw">head</span>(orthologs)</code></pre></div>
<pre><code>##   entry_nr                                        entry_url      omaid
## 1  6284774 https://omadev.cs.ucl.ac.uk/api/protein/6284774/ COCLU03588
## 2  6341447 https://omadev.cs.ucl.ac.uk/api/protein/6341447/ COLGR10929
## 3  6349325 https://omadev.cs.ucl.ac.uk/api/protein/6349325/ COLSU06788
## 4  6410870 https://omadev.cs.ucl.ac.uk/api/protein/6410870/ FUSO411719
## 5  6418086 https://omadev.cs.ucl.ac.uk/api/protein/6418086/ GIBZA02345
## 6  6443712 https://omadev.cs.ucl.ac.uk/api/protein/6443712/ NECHA14651
##   canonicalid                     sequence_md5 oma_group  oma_hog_id
## 1             c5db7c8e6b0eee5c6bbc18fe2dbf9ba7    721679 HOG:0367321
## 2             2cf82b1ffadc581d82d920af9b0130a2    721679 HOG:0367321
## 3  A0A066XI05 d6abbfd6d6ed96217331e86733ae4aeb    721679 HOG:0367321
## 4  A0A0D2XKA6 b92bedd8c4fe61cc3667d4f108ffb528    721679 HOG:0367321
## 5             54814145e0f45beb33ac2e2cb99335a3    721678 HOG:0367321
## 6             8c85e6f457288d146cb99fbc61404cb9    721679 HOG:0367321
##         chromosome locus.start locus.end locus.strand rel_type distance
## 1       scaffold_6       27852     30189           -1      1:1      105
## 2         GG697333      764822    767178            1      1:1      108
## 3  Scaffolds0609.1         399      2755            1      1:1      109
## 4                4     4317479   4319279           -1      1:1      107
## 5  Supercontig_3.1     6701031   6703176           -1      1:1      109
## 6 sca_14_chr10_3_0      669785    671940           -1      1:1      105
##     score
## 1 1362.00
## 2 1320.99
## 3 1312.07
## 4 1326.20
## 5 1327.66
## 6 1336.03</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">formatted_orthologs =<span class="st"> </span><span class="kw">formatData</span>(orthologs)</code></pre></div>
<p>For further information on the OMA REST API please visit (<a href="http://omadev.cs.ucl.ac.uk/api/docs">OMA REST API DOCUMENTATION</a>).</p>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
