<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Klara Kaleb" />

<meta name="date" content="2018-12-21" />

<title>Get started with OmaDB</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Get started with OmaDB</h1>
<h4 class="author"><em>Klara Kaleb</em></h4>
<h4 class="date"><em>2018-12-21</em></h4>



<p>This little vignette shows you how to get started with the <code>OmaDB</code> package. OmaDB is a wrapper for the REST API for the Orthologous MAtrix project (OMA) which is a database for the inference of orthologs among complete genomes.</p>
<p>For more details on the OMA project, see <a href="https://omabrowser.org/oma/home/" class="uri">https://omabrowser.org/oma/home/</a>.</p>
<p>Note that each function in the package has its own individual documentation, which can be accessed by putting a question mark (?) in front of the function name e.g. ?getProtein() .</p>
<div id="some-useful-functions" class="section level2">
<h2>Some useful functions</h2>
<p>The package contains a range of functions that are used to query the database in an R friendly way. This vignette highlights some of them, whereas some others are described in more detail in other vignettes:</p>
<p><a href="exploring_hogs.html">Exploring Hierarchical orthologous groups with roma</a></p>
<p><a href="tree_visualisation.html">Exploring Taxonomic trees with roma</a></p>
<p><a href="sequence_mapping.html">Sequence Analysis with roma</a></p>
<p>Note that all of the vignettes focus on exploring the example responses generated previously, allowing them to build with or without an internet connection. For each example response, the query that generated it is given.</p>
<div id="searchprotein" class="section level3">
<h3>searchProtein</h3>
<p>This function searches the OMA database for entries containing the pattern defined and returns the results in a dataframe. Hence, it is usually a good starting place. Example response, generated via searchProtein(‘MAL’), is below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(OmaDB)

xref =<span class="st"> </span><span class="kw">load</span>(<span class="st">'../data/xref.rda'</span>)

<span class="kw">head</span>(xref)</code></pre></div>
<pre><code>## [1] &quot;xref&quot;</code></pre>
</div>
<div id="getgenomepairs" class="section level3">
<h3>getGenomePairs</h3>
<p>This function serves to obtain the orthologs for 2 whole genomes. The result is a dataframe containing information on each member in the pair and their relationship. Below is the representation of the example response, generated using getGenomePairs(‘YEAST’,‘ASHGO’).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">'../data/pairs.rda'</span>)

<span class="kw">head</span>(pairs)</code></pre></div>
<pre><code>##   entry_1.entry_nr                           entry_1.entry_url
## 1          6618226 https://omabrowser.org/api/protein/6618226/
## 2          6618227 https://omabrowser.org/api/protein/6618227/
## 3          6618228 https://omabrowser.org/api/protein/6618228/
## 4          6618229 https://omabrowser.org/api/protein/6618229/
## 5          6618230 https://omabrowser.org/api/protein/6618230/
## 6          6618231 https://omabrowser.org/api/protein/6618231/
##   entry_1.omaid entry_1.canonicalid             entry_1.sequence_md5
## 1    ASHGO00001              Q75FB7 a9b1a6dc9afb2b02afe8fdf8029b5f22
## 2    ASHGO00002              Q75FB6 5d186037c4dd0a89b34d70d596fac86d
## 3    ASHGO00003              Q75FB5 3a83b276f0c9034f7cf66277e7d6c983
## 4    ASHGO00004              Q75FB4 a8611c3f24ac6599e6a36a2710f6d24a
## 5    ASHGO00005              Q75FB3 07277f0ca66fcb49d175667272f1547f
## 6    ASHGO00006              Q75FB2 2b666569856af5f068e532ade89c1140
##   entry_1.oma_group entry_1.oma_hog_id entry_1.chromosome
## 1                 0     HOG:0393392.4c                  I
## 2            203915        HOG:0200818                  I
## 3            214367        HOG:0200433                  I
## 4            768456  HOG:0387657.2d.3a                  I
## 5            530479     HOG:0397172.3b                  I
## 6            563083     HOG:0201049.3a                  I
##   entry_1.locus.start entry_1.locus.end entry_1.locus.strand
## 1                8108              9067                    1
## 2                9537             12593                    1
## 3               12906             13244                    1
## 4               13713             14846                    1
## 5               16155             19850                    1
## 6               20056             23721                   -1
##   entry_1.is_main_isoform entry_2.entry_nr
## 1                    TRUE          6637770
## 2                    TRUE          6637359
## 3                    TRUE          6637360
## 4                    TRUE          6637767
## 5                    TRUE          6636211
## 6                    TRUE          6636209
##                             entry_2.entry_url entry_2.omaid
## 1 https://omabrowser.org/api/protein/6637770/    YEAST04806
## 2 https://omabrowser.org/api/protein/6637359/    YEAST04395
## 3 https://omabrowser.org/api/protein/6637360/    YEAST04396
## 4 https://omabrowser.org/api/protein/6637767/    YEAST04803
## 5 https://omabrowser.org/api/protein/6636211/    YEAST03247
## 6 https://omabrowser.org/api/protein/6636209/    YEAST03245
##   entry_2.canonicalid             entry_2.sequence_md5 entry_2.oma_group
## 1          RCE1_YEAST 605098a0697ad8fc7af2101e758033cb            494558
## 2          ZDS2_YEAST 8cc75f16fbfd321833abc48fd1173154            203915
## 3          YMK8_YEAST 783fea3b573632292d89a5a5218b6e90            214367
## 4          SCS7_YEAST 9307c7a6e80ed39d8b6529329fee1819            768456
## 5          SMC3_YEAST 4e8f1295434b44ae8f749cad976b966e            530479
## 6          NET1_YEAST f2ba71aea520ea66f015ba357eb6e8c6            563083
##   entry_2.oma_hog_id entry_2.chromosome entry_2.locus.start
## 1     HOG:0393392.4c               XIII              814364
## 2     HOG:0200818.1a               XIII               51640
## 3        HOG:0200433               XIII               54793
## 4  HOG:0387657.2d.3a               XIII              809623
## 5     HOG:0397172.3b                  X              299157
## 6     HOG:0201049.2a                  X              295245
##   entry_2.locus.end entry_2.locus.strand entry_2.is_main_isoform rel_type
## 1            815311                   -1                    TRUE      1:1
## 2             54468                    1                    TRUE      1:1
## 3             55110                    1                    TRUE      1:1
## 4            810777                   -1                    TRUE      1:1
## 5            302849                   -1                    TRUE      1:1
## 6            298814                    1                    TRUE      1:1
##   distance   score
## 1 122.0000  636.04
## 2  95.0000 1424.24
## 3  50.0000  557.51
## 4  37.7442 2682.46
## 5  58.0000 5548.40
## 6  90.0000 1844.35</code></pre>
</div>
<div id="getprotein" class="section level3">
<h3>getProtein</h3>
<p>This function serves to obtain the information for either a single protein entry or multiple protein entries in a database. For more info, see ?getProtein(). There are similar functions to obtain information on genomes, OMA groups and HOGs i.e. getGenome(), getOMAGroup() and getHOG() respectively.</p>
</div>
<div id="getobjectattributes" class="section level3">
<h3>getObjectAttributes</h3>
<p>Single entries in the database are represented as S3 objects, with their attributes corresponding to the information requested. These attributes vary greatly from object to object, and the helper function getObjectAttributes() allows the user to list all the object attributes and their corresponding data types.</p>
</div>
<div id="getattribute" class="section level3">
<h3>getAttribute</h3>
<p>The specific attributes of the created object can be accessed via $ or via the getAttribute() function. Below is an example of object containing information about an OMA group.</p>
<p>Below is the exploration of the example OMA group entry response, obtained via getOMAGroup(‘737636’).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="st">'../data/group.rda'</span>)

object_attributes =<span class="st"> </span><span class="kw">getObjectAttributes</span>(group)</code></pre></div>
<pre><code>## [1] &quot;group_nr : integer&quot;
## [1] &quot;fingerprint : character&quot;
## [1] &quot;related_groups : URL&quot;
## [1] &quot;members : data.frame&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">group<span class="op">$</span>fingerprint</code></pre></div>
<pre><code>## [1] &quot;FPNDKFP&quot;</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getAttribute</span>(group, <span class="st">'fingerprint'</span>)</code></pre></div>
<pre><code>## [1] &quot;FPNDKFP&quot;</code></pre>
</div>
<div id="lazy-loading" class="section level3">
<h3>lazy loading</h3>
<p>In most cases there is great quantity of information available for a given entry and this impacts the data retrival time. Due to this, the information available for such entries is split into a number of endpoints and these are included appropriatelly as redirects in URL form. These are automatically loaded upon $ or getAttribute() accession.</p>
<p>For further information on the OMA REST API please visit <a href="http://omadev.cs.ucl.ac.uk/api/docs">OMA REST API DOCUMENTATION</a>.</p>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
